completion/available/getopt.completion.bash

```diff
@@ -0,0 +1,32 @@
+__getopt() {
+  local OPTIONS=('-a' '--alternative'
+    '-h' '--help'
+    '-l' '--longoptions'
+    '-n' '--name'
+    '-o' '--options'
+    '-q' '--quiet'
+    '-Q' '--quiet-output'
+    '-s' '--shell'
+    '-T' '--test'
+    '-u' '--unquoted'
+    '-V' '--version')
+
+  local SHELL_ARGS=('sh' 'bash' 'csh' 'tcsh')
+
+  local current=$2
+  local previous=$3
+
+  case $previous in
+    -s|--shell)
+      readarray -t COMPREPLY < <(compgen -W "${SHELL_ARGS[*]}" -- "$current")
+      ;;
+    -n|--name)
+      readarray -t COMPREPLY < <(compgen -A function -- "$current")
```

Comment:
So looking at the [manpage]() for getopt I see the following for `-n` :
```
-n, --name progname
           The name that will be used by the getopt(3) routines when it
           reports errors. Note that errors of getopt(1) are still
           reported as coming from getopt.
```

Reading this, it doesn't feel like `-A function` covers the intent of the argument?

If anything, `-A command` feels more appropriate.  But just the same, I really don't know if it's useful to show all of the system functions or the commands to the user.  Just feels like it would spam the terminal with no useful data.

As a helper, tool, it feels like just showing something like `"basname(\$0)"` or `"<progname>"` might be the most useful thing to display.

Reply:
Yes, this should be set to exactly `${COMP_WORDS[0]}` to get the name of the program being completed (which would obviously be `getopt`.